src/Aspire.Dashboard/Components/Controls/SpanDetails.razor

```diff
@@ -84,25 +72,17 @@
                 <PropertyGrid TItem="OtlpSpanEvent"
                               Items="@FilteredSpanEvents"
                               GridTemplateColumns="1fr 2fr"
-                              NameColumnValue="(vm) => DurationFormatter.FormatDuration(vm.TimeOffset(ViewModel.Span))"
                               NameColumnTitle="@(Loc[nameof(ControlsStrings.TimeOffsetColumnHeader)])"
-                              ValueColumnValue="(vm) => vm.Name"
                               ValueColumnTitle="@(Loc[nameof(ControlsStrings.EventColumnHeader)])"
                               IsNameSortable="false"
                               IsValueSortable="false"
                               HighlightText="@_filter">
                     <ExtraValueContent>
                         @if (context.Attributes.Length > 0)
                         {
-                            // There's a weird bug where trying to render a nested FluentDataGrid here with a non-primitive
-                            // TItem leads to the click event not being raised *only in the value (not header) rows*. A
-                            // workaround is to use the attribute index as the item and query the attributes list for its
-                            // name and value in the name/value producers.
-                            <PropertyGrid TItem="int"
-                                          Items="@context.Attributes.Select((_, i) => i).AsQueryable()"
+                            <PropertyGrid TItem="OtlpSpanAttributeItem"
+                                          Items="@context.Attributes.Select(pair => new OtlpSpanAttributeItem(pair.Key, pair.Value)).AsQueryable()"
```

Comment:
@JamesNK can you help me find a way to validate that the bug described here isn't occurring? I stuck a breakpoint in the lambda here but couldn't get it to hit in TestShop. Is there a better playground to use to test this?

Reply:
@JamesNK do you recall how you determined that using a value type worked? Was there some error message you saw or analysis technique you employed?

Reply:
Pushed a fix for this.